pwndbg/arch.py

```diff
@@ -29,6 +30,21 @@ def fix_arch(arch):
     if 'arm' in arch:
         return 'armcm' if '-m' in arch else 'arm'
 
+    # handle some corner cases
+    if arch == 'i386:x86-64':
+        return 'x86-64'
+
+    return arch
+
+def get_init_arch():
+    out = gdb.execute("show architecture", to_string=True).strip()
```

Comment:
I assume with this we will get a better debugging experience if we do gdb ./vmlinux? or is there another story?

Reply:
The original implementation was wrong. It hardcoded an i386 architecture before attach remote. This caused some troubles during kernel debugging. So what I did here is to pull the architecture info from gdb during initialization before attach remote. Once attach remote is issued, we trust its handlers